File: lab01/builder/src/main/java/com/example/builder/client/CharacterBuildClient.java
package com.example.builder.client;

import com.example.builder.builder.CharacterBuilder;
import com.example.builder.director.CharacterDirector;
import com.example.builder.dto.CharacterBuildRequest;
import com.example.builder.model.Character;

public class CharacterBuildClient {
    private final CharacterDirector director;
    private CharacterBuilder builder;

    public CharacterBuildClient() {
        this.director = new CharacterDirector();
    }

    public Character buildCharacter(CharacterBuildRequest request) {

        this.builder = request.getType().createBuilder();

        director.setBuilder(builder);

        return director.make(request);
    }
}

File: lab01/builder1/Client/CharacterBuildClient.cs
using Builder1.Director;
using Builder1.Dto;
using Builder1.Enums;
using Builder1.Models;

namespace Builder1.Client;

public class CharacterBuildClient
{
    private readonly CharacterDirector _director = new();

    public Character BuildCharacter(CharacterBuildRequest request)
    {
        var builder = request.Type.CreateBuilder();
        _director.SetBuilder(builder);
        return _director.Make(request);
    }
}
